### PR TITLE
Multipage Surveys

### PR DESCRIPTION
**Pull request by KaranSampath**
Working on ensuring that the numbering of questions works properly. The rest of the implementation should be correct.

### CODE REVIEW

**Comment by markwhiting** [comment]
@KaranSampath looks good overall. Assign me as a reviewer once you're ready for that. 

Also I notice that the failing test (`Missing CSRF middleware`) is requesting we add back CSRF. @sumants-dev or @KaranSampath I thought we had left it out because it is now built in to Express, but perhaps we should explicitly put it back?

**Comment by sumants-dev** [comment]
I removed it because it added some complications. I added it back. @KaranSampath  If you have question how it works, DM me, I can go into it.

@KaranSampath  Also, the final page submission doesn't seem to work. On the last page of the survey for both paged and multipage. It just returns to the survey. You need to add this commented part back for the last page submission.

``` // if (admin) {
  //   res.render("thanks", {
  //     code: JSON.stringify(req.body, null, 2),
  //     admin: admin,
  //   });
  // } else res.redirect("/"); // Needs to be updated to deal with multiple page surveys```

**Comment by markwhiting** [comment]
Thanks @sumants-dev and makes sense re CSRF.

**Comment by KaranSampath** [comment]
Ah, thanks for pointing out 1, didn't see that and will get that fixed.

On 2, I thought about adding next but realized that since workers aren't going to be able to access previous pages it may be slightly misleading for them. Won't just a submit make it clear that these answers couldn't be changed later on?

**Comment by markwhiting** [comment]
If you're thinking about how to make it more clear that people can't go back, perhaps we do something like "Submit page x of y"?

**Comment by KaranSampath** [comment]
@sumants-dev I'm slightly unsure how to fix the submission issue, it seems my conditionals aren't working and I'm unsure why. Could you please take a quick look at it? Thanks!

Have fixed the other issues.

**Comment by sumants-dev** [comment]
You forgot to pipe the variables of check and final in survey.pug. From my testing it was fixed. Check on your end also. Best,Sumant From: Karan SampathSent: Monday, November 15, 2021 1:16 PMTo: Watts-Lab/surveyorCc: Sumant R Shringari; MentionSubject: Re: [Watts-Lab/surveyor] Multipage Surveys (PR ***@***.*** I'm slightly unsure how to fix the submission issue, it seems my conditionals aren't working and I'm unsure why. Could you please take a quick look at it? Thanks!Have fixed the other issues.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by KaranSampath** [comment]
Tested and fixed numbering. The rendering of the survey looks different so let me know if that's okay otherwise I think the PR is ready to merge @markwhiting.

File: views/survey.pug [hunk 1]
```diff
@@ -21,7 +21,7 @@ html
               else if survey[index-1].group_prompt != item.group_prompt
                 h2= item.group_prompt ? item.group_prompt : generic_group_prompt
               if item.prompt
-                li(class='inputBlock')
+                li(class='inputBlock')= start++
```
Comment (markwhiting) [inline views/survey.pug:24]: might be better to put this number as part of the h4 so it doesn't wrap lines. Also trail with a `.`, so it looks like a number. 

e.g., 
> 3. Select the word that best describes what the person in the picture is thinking or feeling

**Review by markwhiting — CHANGES_REQUESTED** [review]
Overall looks great. I just think we could have a minor tweak to get the question format more consistent but otherwise seems good to go.

**Review by markwhiting — APPROVED** [review]
👍

**Pull request merged**